api/src/API/Controllers/OrganizationsController.cs

```diff
@@ -43,11 +49,15 @@ public async Task<IActionResult> GetOneOrganization(string orgId)
         }
 
         [HttpPost]
-        public async Task<IActionResult> CreateNewOrganization(Organization organization)
+        public async Task<IActionResult> CreateNewOrganization([FromBody] CreateOrganizationRequest body)
         {
-            organization.Id = Guid.NewGuid();
+            var organization = body.Organization;
             OrganizationContainerClient container = containerProvider.OrganizationContainer;
             var addedOrg = await container.AddOneAsync(organization);
+
+            var secretName = addedOrg.Id + "-client-secret";
+            await this.keyVaultClient.PutSecretAsync(secretName, body.ClientSecret);
```

Comment:
I don't think we should do this in code. It's a one time configuration of the key vault secret.

Reply:
Secrets are per org

Reply:
Yes?

Reply:
How do you suggest adding the secret when we create a new org if we don't do it here?

Reply:
Org creation process isn't something that happens outside of our control like race result submissions are. It's more like org "onboarding" more than anything else. This relates to org models containing a wildapricot client id as a required property. A generic organization doesn't inherently have a wild apricot client id. I think we should either uniformly approach the design for scalability to other orgs or we shouldn't at all. Something in between is unproductive because then the work done doesn't actually apply to other orgs.

Reply:
Nothing we have designed so far scales with that model. The frontend and backend do not fit to that model. We would need to have logic specific to organization. The frontend would need to ask the backend "what endpoints do I need to know about for this org?"